flink-runtime/src/test/java/org/apache/flink/runtime/io/network/netty/NettyClientServerSslTest.java

```diff
@@ -65,6 +68,60 @@ public void testValidSslConnection() throws Exception {
 
 		Channel ch = NettyTestUtil.connect(serverAndClient);
 
+		SslHandler sslHandler = (SslHandler) ch.pipeline().get("ssl");
+		assertTrue("default value should not be propagated", sslHandler.getHandshakeTimeoutMillis() >= 0);
+		assertTrue("default value should not be propagated", sslHandler.getCloseNotifyTimeoutMillis() >= 0);
+
+		// should be able to send text data
+		ch.pipeline().addLast(new StringDecoder()).addLast(new StringEncoder());
+		assertTrue(ch.writeAndFlush("test").await().isSuccess());
+
+		NettyTestUtil.shutdown(serverAndClient);
+	}
+
+	/**
+	 * Verify valid (advanced) ssl configuration and connection.
+	 */
+	@Test
+	public void testValidSslConnectionAdvanced() throws Exception {
```

Comment:
This is quite poor test :( With respect to `SESSION_CACHE_SIZE` and `SESSION_TIMEOUT` it tests only for "not throwing any exception". If those properties are just ignored, the test will still pass. 

Can we add some stress test that actually verifies the bug which this PR is trying to solve? Maybe stress test AND benchmark like `StreamNetworkThroughputBenchmarkTest#largeRemoteMode`?

Reply:
Actually, this test verifies that the session cache size and session timeout are set in Netty's ssl handler and that should be enough. Whether they are actually being used should be tested in Netty's tests.

Reply:
Yes, you are right regarding `handshake-timeout` and `close-notify-flush-timeout`, but as I wrote previously, I do not see how `SESSION_CACHE_SIZE` and `SESSION_TIMEOUT` are tested at all. And regardless of that, it still would be better to add a stress test/benchmark for that. Depends how important this feature is... However if it's not important one could argue why even bother supporting this? 

On a side note. Couldn't we provide some generic way to configure netty? Like passing any config option prefixed `taskmanager.netty.client` to taskmanager's netty client, without manually specifying and handling them by us?

Reply:
Actually, I found a way to verify that these two properties are also set - will update the test.

Do you think, a benchmark should be included in this PR or rather separately (it is not really related to these changes)? Also: only in `master`?

About the `taskmanager.netty.client` prefix: that sounds like a nice idea and could probably be extended to similar use cases with component-specific configurations. If you think, it's worth pursuing, can you open a JIRA ticket for this improvement?